packages/world-modules/src/modules/callwithsignature/getSignedMessageHash.sol

```diff
@@ -34,7 +37,9 @@ function getSignedMessageHash(
       abi.encodePacked(
         "\x19\x01",
         domainSeperator,
-        keccak256(abi.encode(CALL_TYPEHASH, signer, systemId, keccak256(callData), nonce))
+        keccak256(
+          abi.encode(CALL_TYPEHASH, signer, keccak256(abi.encode(systemId.toString())), keccak256(callData), nonce)
```

Comment:
I think we want to do a straight cast here, not `toString`, as `toString` is more meant as the human-readable version for logging/debugging (e.g. `sy:<root>:Register`)

Reply:
Oh I didn't think about that! thanks